src/System.Collections/src/System/Collections/Generic/SortedList.cs

```diff
@@ -161,11 +161,11 @@ public SortedList(IDictionary<TKey, TValue> dictionary, IComparer<TKey> comparer
                 Debug.Assert(count == _keys.Length);
                 if (count > 1)
                 {
-                    Array.Sort<TKey, TValue>(keys, _values, comparer);
-                    comparer = Comparer; // obtain default if this is null.
+                    // Comparer is set to the default if the comparer parameter is null
+                    Array.Sort<TKey, TValue>(keys, _values, Comparer);
                     for (int i = 1; i != keys.Length; ++i)
                     {
-                        if (comparer.Compare(keys[i - 1], keys[i]) == 0)
+                        if (Comparer.Compare(keys[i - 1], keys[i]) == 0)
```

Comment:
I'm not at all sure about the range check. I have come to recognize this as a problem area, though. The JIT has to reason that `i >= 1` and `i - 1` always `>= 0´. This would be doable in a generalized data flow framework but given the JITs current level of abilities this seems out of range.

            var array1 = GetArray(1000);

            var sum = 0;
            for (int i = 1; i < array1.Length; i++)
            {
                if (array1[i - 1] == array1[i])
                    sum++;
            }

            Sink(sum);

Here, the `i - 1` access range checks on Desktop 4.6.1. `i` is range check free.

Changing to `i != array1.Length` range checks both. Changing to `++i` does not change anything.